File: Difficulty: Easy/Deletion at the end of a Linked List/deletion-at-the-end-of-a-linked-list.py
'''
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
'''
class Solution:
    def removeLastNode(self, head):
        # code here
        if not head:
            return None
        if not head.next:
            return None
        second_last = head
        while second_last.next.next:
            second_last = second_last.next
        second_last.next = None
        return head
        

File: Difficulty: Easy/Deletion at the end of a Linked List/deletion-at-the-end-of-a-linked-list.cpp
/*
class Node {
public:
    int data;
    Node* next;

    Node(int x) {
        data = x;
        next = NULL;
    }
};
*/
class Solution {
  public:
    Node* removeLastNode(Node* head) {
        // code here
        if(head == NULL)
        return head;
        
        if(head->next == NULL)
        return NULL;
        
        Node* p= NULL, *q = head;
        while(q->next != NULL){
            p=q;
            q = q->next;
        }
        p->next= NULL;
        free(q);
        return head;
    }
};